src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -341,44 +559,60 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
         eirModifierFuncPLR = 0.0;
     }
 
-    this->powerUsage = (this->loadSideHeatTransfer / this->referenceCOP) * eirModifierFuncPLR * eirModifierFuncTemp;
+    this->powerUsage =
+        (availableCapacity / this->referenceCOP) * eirModifierFuncPLR * eirModifierFuncTemp * InputPowerMultiplier * this->cyclingRatio;
     this->powerEnergy = this->powerUsage * reportingInterval;
 
     // energy balance on heat pump
     this->sourceSideHeatTransfer = this->calcQsource(this->loadSideHeatTransfer, this->powerUsage);
     this->sourceSideEnergy = this->sourceSideHeatTransfer * reportingInterval;
 
     // calculate source side outlet conditions
-    Real64 CpSrc = 0.0;
+    Real64 CpSrc;
     if (this->waterSource) {
-        CpSrc = FluidProperties::GetSpecificHeatGlycol(state,
-                                                       thisLoadPlantLoop.FluidName,
-                                                       state.dataLoopNodes->Node(this->loadSideNodes.inlet).Temp,
-                                                       thisLoadPlantLoop.FluidIndex,
-                                                       "PLHPEIR::simulate()");
-    } else if (this->airSource) {
+        auto &thisSourcePlantLoop = state.dataPlnt->PlantLoop(this->sourceSidePlantLoc.loopNum);
+        CpSrc = FluidProperties::GetSpecificHeatGlycol(
+            state, thisSourcePlantLoop.FluidName, this->sourceSideInletTemp, thisSourcePlantLoop.FluidIndex, "PLHPEIR::simulate()");
+    } else {
         CpSrc = Psychrometrics::PsyCpAirFnW(state.dataEnvrn->OutHumRat);
     }
     Real64 const sourceMCp = this->sourceSideMassFlowRate * CpSrc;
     this->sourceSideOutletTemp = this->calcSourceOutletTemp(this->sourceSideInletTemp, this->sourceSideHeatTransfer / sourceMCp);
+
```

Comment:
this next code was to prevent souce side temps from getting out of control. code above replaces this but this was not removed in case that other code wasn't working correctly. We saw some pretty out of control temps when dispatching the heat recovery HP, on the source side, because load side was controlling and source side just put out heat to the source plant.